wire-ios/Wire-iOS/Sources/AnalyticsTrackingManager/TrackingManager.swift

```diff
@@ -71,14 +101,29 @@ final class TrackingManager: TrackingInterface {
 
     func enableAnalytics() async throws {
         try await sessionManager.makeEnableAnalyticsUseCase()?.invoke()
-        ExtensionSettings.shared.disableAnalyticsSharing = false
         AVSFlowManager.getInstance()?.setEnableMetrics(true)
+        journal?[.isAnalyticsTrackingConsentGiven] = true
     }
 
     func disableAnalytics() throws {
         try sessionManager.makeDisableAnalyticsUseCase()?.invoke()
-        ExtensionSettings.shared.disableAnalyticsSharing = true
         AVSFlowManager.getInstance()?.setEnableMetrics(false)
+        journal?[.isAnalyticsTrackingConsentGiven] = false
+    }
+
+    /// Previously the consent for analytics tracking was stored only once per app.
+    /// If the consent has been given, mark all currently set up accounts as consent being given.
+
+    private func migrateFromLegacyStorageIfNeeded() {
+        guard let disableAnalyticsSharing = ExtensionSettings.shared.disableAnalyticsSharing else { return }
+
+        for account in sessionManager.accountManager.accounts {
+            let userID = account.userIdentifier
+            let journal = Journal(userID: userID, storage: sharedUserDefaults)
+            journal[.isAnalyticsTrackingConsentGiven] = !disableAnalyticsSharing
+        }
+
+        ExtensionSettings.shared.disableAnalyticsSharing = nil
     }
```

Comment:
suggestion: use appVersionMigration for this. so it's executed only once. Also then we can clean previous setup.

question: is ExtensionSettings.shared.disableAnalyticsSharing obsolete then?

Reply:
yes, removed `ExtensionSettings.shared.disableAnalyticsSharing`